Mathlib/Algebra/Polynomial/Splits.lean

```diff
@@ -322,6 +322,23 @@ theorem eq_prod_roots_of_splits_id {p : K[X]} (hsplit : Splits (RingHom.id K) p)
     p = C p.leadingCoeff * (p.roots.map fun a => X - C a).prod := by
   simpa using eq_prod_roots_of_splits hsplit
 
+theorem Splits.dvd_of_roots_le_roots {p q : K[X]} (hp : p.Splits (RingHom.id _)) (hp0 : p ≠ 0)
+    (hq : p.roots ≤ q.roots) : p ∣ q := by
+  by_cases hq0 : q = 0
+  · simp [hq0]
```

Comment:
```suggestion
  rcases eq_or_ne q 0 with rfl | hq0
  · exact dvd_zero p
```

Reply:
In my opinion my proof is more readable. I don't mind either of these proofs though.

Reply:
IMHO, this falls into "whatever the author prefers" bin, not into "we enforce one of the styles" bin.